### DISCUSSION TITLE
Flux.@functor errors

### DISCUSSION THREAD

**Discussion by BCRARL** [fonsp/Pluto.jl#1380] [Pluto development] [answered]
This cell didn't work in a a brand new notebook and I can't see why it shouldn't
```
begin
	using Flux
	struct Reshape
	   shape
        end; 
	Reshape(args...) = Reshape(args); 
	(r::Reshape)(x) = reshape(x, r.shape);
	Flux.@functor Reshape ()
end
```

The response is
```
LoadError: UndefVarError: Flux not defined
in expression starting at .julia/pluto_notebooks/Cute program.jl#==#3623c510-f95d-11eb-149e-c99cf971ce14:8
top-level scope@:0
```

**Comment by Pangoraw** [comment, marked as answer]
This is the same problem as https://github.com/fonsp/Pluto.jl/issues/1340#issuecomment-888353425. You can solve it by moving `using Flux` to its own cell.

**Reply by BCRARL** [reply]
Except that the split followed by
```
using BSON
```
 and then loading a Flux model with
```
BSON.@load "path" Model
```
leads to 
```
UndefVarError: Flux not defined
```

**Reply by BCRARL** [reply]
That's what I call a catch 22

**Reply by Pangoraw** [reply]
You should use `BSON.load("path", @__MODULE__)` because the `using Flux` is not in `Main` but in `workspaceXXX` (due to Pluto internals). You can help Pluto determine the execution order by adding a dummy reference to `Reshape` in the loading cell:
```julia
begin
    Reshape # to make sure Reshape is defined before this cell is run
    model = BSON.load("path", @__MODULE__)[:model]
end
```

**Reply by BCRARL** [reply]
At this point I am manually building the execution tree with hacks. I guess Pluto is not the right solution to my development needs.

**Reply by fonsp** [reply]
@BCRARL opened a new discussion about this here: https://github.com/fonsp/Pluto.jl/discussions/1383 Closing this
